src/main/java/org/testng/xml/internal/TestNamesMatcher.java

```diff
@@ -0,0 +1,106 @@
+package org.testng.xml.internal;
+
+import java.util.Iterator;
+import java.util.List;
+import org.testng.TestNGException;
+import org.testng.collections.Lists;
+import org.testng.xml.XmlSuite;
+import org.testng.xml.XmlTest;
+
+/**
+ * The class to work with "-testnames"
+ *
+ */
+public final class TestNamesMatcher {
+    
+    private final List<XmlSuite> cloneSuites = Lists.newArrayList();
+    private final List<String> matchedTestNames = Lists.newArrayList();
+    private final List<XmlTest> matchedTests = Lists.newArrayList();
+    private List<String> testNames = Lists.newArrayList();
+    
+    /**
```

Comment:
Nitpicking : This javadoc doesn't add any value. We might as well, just remove this off.

Reply:
Done.